File: src/core/helm/model/Chart.java
/*
 * Copyright (C) 2023 Hedera Hashgraph, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.hedera.fullstack.helm.client.model;

/**
 * Represents a chart and is used to interact with the Helm install and uninstall commands.
 * @param repoName the name of repository which contains the Helm chart.
 * @param name the name of the Helm chart.
 */
public record Chart(String name, String repoName) {
    public Chart(String name) {
        this(name, null);
    }

    @Override
    public String toString() {
        if (repoName == null || repoName.isBlank()) {
            return name;
        }

        return String.format("%s/%s", repoName, name);
    }

    public String qualified() {
        return toString();
    }

    public String unqualified() {
        return name;
    }
}

File: src/integration/helm/model/chart.ts
// SPDX-License-Identifier: Apache-2.0

/**
 * Represents a chart and is used to interact with the Helm install and uninstall commands.
 */
export class Chart {
  /**
   * Creates a new Chart instance.
   * @param name the name of the Helm chart.
   * @param repoName the name of repository which contains the Helm chart.
   */
  constructor(
    public readonly name: string,
    public readonly repoName?: string,
  ) {}

  /**
   * Returns a string representation of the chart.
   * If repoName is provided, returns "repoName/name", otherwise just returns "name".
   */
  toString(): string {
    if (!this.repoName?.trim()) {
      return this.name;
    }
    return `${this.repoName}/${this.name}`.replace(/\/$/, '');
  }

  /**
   * Returns the qualified name of the chart (same as toString).
   */
  qualified(): string {
    return this.toString();
  }

  /**
   * Returns the unqualified name of the chart (just the name without the repo).
   */
  unqualified(): string {
    return this.name;
  }
}
